Tests/CMakeLib/run_link_commands.cpp

```diff
@@ -0,0 +1,164 @@
+#include "cmConfigure.h" // IWYU pragma: keep
+
+#include <cstdlib>
+#include <iostream>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "cmsys/FStream.hxx"
+
+#include "cmSystemTools.h"
+
+class LinkCommandParser
```

Comment:
Why you don't use existing JSON parsers? See https://github.com/nlohmann/json for example.

Reply:
Yes, that's reasonable. Please take a look at json utilities https://github.com/Software-Analysis-Team/CMake/tree/master/Utilities/cmjsoncpp

Reply:
I am not sure nlohmann/json could be easily approved in cmake, but please consider using an existing json parser